app/models/location.rb

```diff
@@ -92,6 +92,14 @@ class Location < ApplicationRecord
     .where.not(id: activating_soon.pluck(:id))
   }
 
+  scope :active_on_date, lambda { |date = Time.zone.now|
```

Comment:
Had I not wanted to also include locations deactivating_soon in the check (as well as active locations) this function would not have been needed and I could have only modified active_status with the optional date.

Reply:
This looks good! You can leave it like this or have something like 
`locations.active_on_date(date).exists? || locations.deactivating_soon(date).exist?`

I think I prefer it the way you did it here, cause it feels that a distinct scope for what is actually active is more reusable than `active_status` which I think we're only using for tags/filtering